parcel-bundler/parcel
# Allow complete urls for publicUrl in server

This allows for the built-in server to host assets at the path portion of a complete url provided as publicUrl.

This PR also includes adding corresponding mime types as `Content-Type` headers in server responses.

Test Plan:
* Temporarily merge #3494, then
* Run the react-hmr example, passing `--public-url="http://localhost:3000/foo" and verify that the script tags in the document served are rooted there. This means that these resources will 404 with the default port of 1234.
* Change the public url to `http://localhost:1234/foo` and verify that resources are served.
* Verify js resources are served with an `application/javascript` `Content-Type` header
* Verify standard path of `--public-url="/foo"` continues to function.
